commitizen/commands/init.py

```diff
@@ -13,59 +13,150 @@
         self.cz = factory.commiter_factory(self.config)
+        self.project_info = ProjectInfo()
 
     def __call__(self):
         if self.config.path:
             out.line(f"Config file {self.config.path} already exists")
             return
 
-        # No config for commitizen exist
-        config_path = self._ask_config_path()
+        out.info("Welcome to commitizen!\n")
+        out.line(
+            "The prompts will ask you different questions " "to configure your project."
+        )
+        out.line("For further configuration visit:")
+        out.line("\n\thttps://commitizen-tools.github.io/commitizen/config/\n")
+
+        # Collect information
+        try:
+            config_path = self._ask_config_path()  # select
+            cz_name = self._ask_name()  # select
+            version_provider = self._ask_version_provider()  # select
+            tag = self._ask_tag()  # confirm & select
+            version = Version(tag)
+            tag_format = self._ask_tag_format(tag)  # confirm & text
+            version_type = self._ask_version_type()  # select
+            update_changelog_on_bump = self._ask_update_changelog_on_bump()  # confirm
+            major_version_zero = self._ask_major_version_zero(version)  # confirm
+        except KeyboardInterrupt:
+            raise InitFailedError("Stopped by user")
+
+        # Initialize configuration
         if "toml" in config_path:
             self.config = TomlConfig(data="", path=config_path)
         elif "json" in config_path:
             self.config = JsonConfig(data="{}", path=config_path)
         elif "yaml" in config_path:
             self.config = YAMLConfig(data="", path=config_path)
-        self.config.init_empty_config_content()
-
         values_to_add = {}
-        values_to_add["name"] = self._ask_name()
-        tag = self._ask_tag()
-        values_to_add["version"] = Version(tag).public
-        values_to_add["tag_format"] = self._ask_tag_format(tag)
-        self._update_config_file(values_to_add)
+        values_to_add["name"] = cz_name
+        values_to_add["tag_format"] = tag_format
+        values_to_add["version_type"] = version_type
 
+        if version_provider == "commitizen":
+            values_to_add["version"] = version.public
+        else:
+            values_to_add["version_provider"] = version_provider
+
+        if update_changelog_on_bump:
+            values_to_add["update_changelog_on_bump"] = update_changelog_on_bump
+
+        if major_version_zero:
+            values_to_add["major_version_zero"] = major_version_zero
+
+        # Collect hook data
         hook_types = questionary.checkbox(
             "What types of pre-commit hook you want to install? (Leave blank if you don't want to install)",
             choices=[
-                questionary.Choice("commit-msg", checked=True),
-                questionary.Choice("pre-push", checked=True),
+                questionary.Choice("commit-msg", checked=False),
```

Comment:
May I know why do we want to set both of them to `false` instead of `true`

Reply:
I think `.pre-commit` with `cz check` is overkill for many projects, I actually don't like enforcing conventional commits because it allows skipping those commits which do not match. Having it set to true feels "enforced" by commitizen. I'd like to run `cz init` press enter many times and only have `commitizen` configured fast. If a set of users make use of `.pre-commit`, then they have to actively set it to true.

Reply:
Got it. I think I'm the so-called "a set of users" haha 😆 But ya, I'm good with it as well 🙂